src/apps/tools/private.clj

```diff
@@ -69,7 +78,7 @@
                       restrict-private-tool
                       (assoc :implementation (ensure-default-implementation user implementation))
                       persistence/add-tool)]
-      (containers/add-tool-container tool-id (restrict-private-tool-container container))
+      (containers/add-tool-container tool-id (restrict-private-tool-container (set-private-tool-defaults container)))
```

Comment:
The basic gist here is: `set-private-tool-defaults` should set anything that isn't provided to the (lower) default limits, where `restrict-private-tool-container` has been repurposed to restrict things down to the higher maximum limits. I've kept it so the latter will still default the values to the default limits, but I wanted to be slightly more explicit, particularly because ultimately we may want to move `restrict-private-tool-container` to another namespace and apply it to all tools, not just private ones.

Reply:
The original intent was to restrict private tools to a maximum of the `apps.tools.private.memory-limit` config (16GB), whether the user provided no limit, or if the provided limit was higher than this configured max.

So is it now the intent to allow the user to set a higher limit for private tools, up to the `apps.tools.memory-limit` max for all tools (32GB)? Or do we still want to restrict private tools to a lower limit than public tools (of course, an  admin should still be able to lift the limit to the `apps.tools.memory-limit` max)?

Reply:
That all tools would be subject to the same effective maximum was my understanding, yeah -- with the prior limits for private tools becoming the defaults instead (and we'll also be hiding this slightly behind something saying it's advanced behaviour). Not much benefit to adding an interface for this unless it can be adjusted both up and down, and I don't think we want to bring the defaults down (though we could -- VICE defaults to 2 cores and I forget how much memory, but I think less than the current private tool defaults). It's also worth noting that the trajectory here is towards this being only the max, adding a field also for the minimum (default whatever value we consider acceptable as the smallest slot), and then allowing users to select a value between them at runtime. Once again there, we'd probably peg the defaults somewhere like these older values, I think